lib/mutant/ast/regexp/transformer.rb

```diff
@@ -0,0 +1,82 @@
+module Mutant
+  module AST
+    module Regexp
+      class Transformer
+        include AbstractType
+
+        REGISTRY = {} # rubocop:disable MutableConstant
```

Comment:
> @backus although I am not entirely sure I understand what the separation between `Registry` and `TypeRegistry` is in the example above. Given that `Registry` has no other state besides a `TypeRegistry` instance I'm not sure why you don't collapse them down and have a single `Registry` object that you interact with.

So the difference is the `TypeRegistry` is just responsible for mapping types to objects. It fails if you try to register something already registered. It fails if you try to register a key that is not a valid ast type. `Mutant::Mutator::Registry` builds on top and supplies the `call` method which does a `lookup` on a node type and then calls the mutator. This was the interface before which is why I preserved it